src/precompiles/sgxattest.rs

```diff
@@ -151,6 +158,32 @@ fn sgxattest_random(_input: &[u8], gas_limit: u64) -> PrecompileResult {
     }
 }
 
+fn sgxattest_httpscall(_input: &[u8], gas_limit: u64) -> PrecompileResult {
+    let gas_used = 10000 as u64;
+    if gas_used > gas_limit {
+        return Err(PrecompileError::OutOfGas);
+    } else {
+
+        let url = String::from_utf8_lossy(_input).into_owned();
+        // let url = "https://dkg.alliance.dev";
+
+        // Decode the ABI input
+        // let decoded = decode(&[ParamType::String], _input).expect("failed to decode");
+        // let url = decoded[0].clone().into_string().expect("failed to decode");
+
+        // Perform the HTTP call
+        let client = Client::new();
+        let response = client.get(url).send().expect("failed to decode");
```

Comment:
Out of curiosity, for which use case do we require the http get request for a given url as a precompile?

Reply:
In a nutshell, we would like a secure TLS/HTTPS oracle we can query, for example to fetch a price from a website or more generally to retrofit existing web2 services with some smart contract power 
Here is the project pitch slides for the hackathon project we worked on that led to this draft https://docs.google.com/presentation/d/1K_YHJfAa_Ar1S7ESCsjaZt6dR9OD_JFMy2dJ_UsQOJA/edit#slide=id.p

Reply:
Oh nice, thanks, I sent an access request to the slides link :)